src/lib/transformers/tokend.js

```diff
@@ -189,6 +173,12 @@ class TokendTransformer {
         transformedProperties = transformedProperties.mergeDeep(value);
       });
 
+      Object.keys(this._cache)
+        .filter((propertyName) => seenProperties.indexOf(propertyName) === -1)
+        .forEach((entry) => {
```

Comment:
You can probably consolidate this line and the previous one within a single `forEach` (or even a regular `for` loop) with an `if` statement. Both `filter` and `forEach` are `O(n)` so you're iterating over the full list, then a subset just to remove values from the array when the time spent in each iteration grows linearly based on the number of properties in the property-set.

Reply:
All set